src/metabase/api/pulse.clj

```diff
@@ -103,19 +107,40 @@
    archived      (s/maybe s/Bool)
    parameters    [su/Map]}
   ;; do various perms checks
-  (validation/check-has-general-permission :subscription false)
-  (let [pulse-before-update (api/write-check Pulse id)]
+  (try
+   (validation/check-has-general-permission :monitoring)
+   (catch clojure.lang.ExceptionInfo _e
+     (validation/check-has-general-permission :subscription false)))
+
+  (let [pulse-before-update (api/write-check (pulse/retrieve-pulse id))]
     (check-card-read-permissions cards)
     (collection/check-allowed-to-change-collection pulse-before-update pulse-updates)
+
+    ;; if advanced-permissions is enabled, only superuser or non-admin with subscription permission can
+    ;; update pulse's recipients
+    (when (premium-features/enable-advanced-permissions?)
+      (let [[_ to-add-recipients] (data/diff (set (map :id (:recipients (api-alert/email-channel pulse-before-update))))
+                                             (set (map :id (:recipients (api-alert/email-channel pulse-updates)))))
```

Comment:
nit: you could also use `clojure.set/difference` which is a bit more specific than `data/diff`